packages/arb-token-bridge-ui/src/token-bridge-sdk/utils.ts

```diff
@@ -0,0 +1,12 @@
+import { Signer } from 'ethers'
+import { Provider } from '@ethersproject/providers'
+
+export const getAddressFromSigner = async (signer: Signer) => {
+  const address = await signer.getAddress()
+  return address
+}
+
+export const getChainIdFromProvider = async (provider: Provider) => {
```

Comment:
If we have provider, we most likely have the corresponding chainId no?

Reply:
Yes in UI we do, but the SDK has been created from the POV of having the bare minimum parameters passed to it. Just pass the `provider` to sdk and we will internally derive all the stuff that the methods require - `chainId`, `network` etc.